Client/Frontend/Browser/TabScrollController.swift

```diff
@@ -252,9 +263,18 @@ extension TabScrollingController: UIScrollViewDelegate {
 
         suppressToolbarHiding = false
     }
-
+    
+    func scrollViewDidEndDecelerating(scrollView: UIScrollView) {
+        showOrHideWebViewContainerToolbar()
+    }
+    
+    func scrollViewDidEndZooming(scrollView: UIScrollView, withView view: UIView?, atScale scale: CGFloat) {
+        showOrHideWebViewContainerToolbar()
+    }
+    
```

Comment:
This seems like it's been an issue for a while - not with just this PR. I don't think we handle adjusting the toolbars properly whenever the user zooms. A potential fix could be to add an observer on the zoomScale property and update the toolbar position whenever it changes.